src/az-purview/deploy/terraform/modules/keyvault/main.tf

```diff
@@ -0,0 +1,62 @@
+terraform {
```

Comment:
There is a Key Vault module at src/common/infrastructure/terraform/modules/private-key-vault/main.tf.  Is there an opportunity to leverage that module instead of creating a new one?  Is there functionality that should be added to the existing module that is requiring a new Key Vault module for this recipe?

Reply:
It is something I spoke with @promisinganuj but we decided not to do it for this PR. @promisinganuj any thoughts?

Reply:
If you or @promisinganuj create an issue indicating we need to modify the Terraform modules to work across the data and non-data related recipes (aligning on an approach), I'm OK with temporarily using the modules as in this PR.

Reply:
I have raised an issue to refactor the terraform modules #30